apps/studio/data/storage/storage-credentials-create-mutation.ts

```diff
@@ -0,0 +1,48 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { storageCredentialsKeys } from './storage-credentials-keys'
+import { post } from 'data/fetchers'
+
+type CreateStorageCredential = {
+  description: string
+  projectRef?: string
+}
+const createStorageCredential = async ({ description, projectRef }: CreateStorageCredential) => {
+  if (!projectRef) {
+    throw new Error('projectRef is required')
+  }
+
+  const res = await post('/platform/storage/{ref}/credentials', {
+    params: {
+      path: {
+        ref: projectRef,
+      },
+    },
+    body: {
+      description,
+    },
+  })
+
+  return res
+}
+
+type StorageCredentialsCreateMutation = {
+  projectRef?: string
+}
+
+export function useStorageCredentialsCreateMutation({
+  projectRef,
+}: StorageCredentialsCreateMutation) {
+  const queryClient = useQueryClient()
+
+  const keys = storageCredentialsKeys.credentials(projectRef)
+
+  return useMutation({
+    mutationFn: ({ description }: { description: string }) =>
+      createStorageCredential({ description, projectRef }),
+    mutationKey: keys,
```

Comment:
you don't need a key for mutations. Remove this and it will simplify the variables (all variables from the `useStorageCredentialsCreateMutation` go directly through to `mutationFn`. Same for the other mutations.